### PR TITLE
First changes

### PR DESCRIPTION
**Pull request by aarmey**
- Provides a custom differencing function for performance.
- Uses the sorting of cells in the lineage to simplify loop iteration.
- Fixes some typos in the tests (!)
- Simplifies some of the censoring code.

### CODE REVIEW

**Comment by aarmey** [comment]
@Farnazmdi please go through this carefully. If this all looks reasonable, I will go through it once more and then merge.

File: lineage/states/stateCommon.py [hunk 1]
```diff
@@ -50,10 +50,10 @@ def gamma_LL(
...
     glnA = gammaln(x[0])
-    outt = 0.0
+    outt = np.zeros(len(x) - 1)
     for i in range(len(x) - 1):
         gobs = gamma_obs[i] / x[i + 1]
-        outt -= np.dot(
+        outt[i] -= np.dot(
             gammas[i] * time_cen[i],
             (x[0] - 1.0) * np.log(gobs) - gobs - glnA - logX[i + 1],
         )
```
Comment (Farnazmdi) [inline lineage/states/stateCommon.py:59]: sorry it is taking too long. I have been trying to remember how we calculated this likelihood..
Is `outt` the log-likelihood value for each condition (control,conc1, conc2, conc3) ?

File: lineage/states/stateCommon.py [hunk 2]
```diff
@@ -62,12 +62,37 @@ def gamma_LL(
...
     return outt
 
 
+
+def gamma_LL_diff(logX: np.ndarray,
+    gamma_obs: list[np.ndarray],
+    time_cen: list[np.ndarray],
+    gammas: list[np.ndarray],
+):
+    base = gamma_LL(logX, gamma_obs, time_cen, gammas)
```
Comment (Farnazmdi) [inline lineage/states/stateCommon.py:77]:
```suggestion
):
    """Calculating the diff for log likelihood of each parameter with step size = 1e-6."""
    base = gamma_LL(logX, gamma_obs, time_cen, gammas)
```

**Review by Farnazmdi — APPROVED** [review]
Thank you so much. I wonder how the tests were passing...
I added few docstrings for myself to remember.

File: lineage/states/stateCommon.py (hunk 1 above)
Comment (aarmey) [reply lineage/states/stateCommon.py:59]: Yes. I separated the conditions in `outt` because it allows me to do much of the differencing in one call.

**Comment by aarmey** [comment]
Yes, we need to look into the tests. Those arguments were misspelled, so they were essentially only running for the uncensored case. However, I have gone through this again and am confident in the changes. Merging.

**Pull request merged**